src/main/java/com/exasol/adapter/dialects/exasol/ExasolColumnMetadataReader.java

```diff
@@ -52,4 +54,58 @@ public DataType mapJdbcType(final JdbcTypeDescription jdbcTypeDescription) {
             return super.mapJdbcType(jdbcTypeDescription);
         }
     }
+
+    @Override
+    public JdbcTypeDescription readJdbcTypeDescription(final ResultSet remoteColumn) throws SQLException {
+        final JdbcTypeDescription typeDescription = super.readJdbcTypeDescription(remoteColumn);
+        if (typeDescription.getJdbcType() == EXASOL_INTERVAL_DAY_TO_SECONDS) {
+            return extractIntervalDayToSecondPrecision(remoteColumn, typeDescription);
+        } else if (typeDescription.getJdbcType() == EXASOL_INTERVAL_YEAR_TO_MONTHS) {
+            return extractIntervalYearToMonthPrecision(remoteColumn, typeDescription);
+        } else {
+            return typeDescription;
+        }
+    }
+
+    private String getTypeDescriptionStringForColumn(final ResultSet remoteColumn) throws SQLException {
+        try (final PreparedStatement preparedStatement = this.connection.prepareStatement(
+                "SELECT COLUMN_TYPE FROM SYS.EXA_ALL_COLUMNS WHERE COLUMN_SCHEMA = ? AND COLUMN_TABLE = ? AND COLUMN_NAME = ?;")) {
+            final String schema = remoteColumn.getString("TABLE_SCHEM");
+            final String table = remoteColumn.getString("TABLE_NAME");
+            final String column = remoteColumn.getString("COLUMN_NAME");
+            preparedStatement.setString(1, schema);
+            preparedStatement.setString(2, table);
+            preparedStatement.setString(3, column);
+            try (final ResultSet resultSet = preparedStatement.executeQuery()) {
+                resultSet.next();
+                return resultSet.getString("COLUMN_TYPE");
+            }
+        }
+    }
+
+    private JdbcTypeDescription extractIntervalDayToSecondPrecision(final ResultSet remoteColumn,
+            final JdbcTypeDescription typeDescription) throws SQLException {
+        final String typeDescriptionString = getTypeDescriptionStringForColumn(remoteColumn);
+        final Pattern pattern = Pattern.compile("INTERVAL DAY\\((\\d+)\\) TO SECOND\\((\\d+)\\)");
+        final Matcher matcher = pattern.matcher(typeDescriptionString);
+        if (matcher.matches()) {
+            return new JdbcTypeDescription(typeDescription.getJdbcType(), Integer.parseInt(matcher.group(2)),
+                    Integer.parseInt(matcher.group(1)), typeDescription.getByteSize(), typeDescription.getTypeName());
+        } else {
+            throw new IllegalStateException("Failed to extract INTERVAL precision");
```

Comment:
How about using the default values we used before? It will be more user-friendly

Reply:
This also works. (I just added an integration test for this case). The database generates the type definition with the precision also for the default type.
The exception will only be thrown if the database changes the syntax of the type definition.